### ISSUE TITLE
Pyface minor version release 7.1.0

### ISSUE THREAD

**Issue by rahulporuri** [enthought/pyface#714]
Target date: To be confirmed, will edit here once confirmed.
This is a minor release. See https://github.com/enthought/traitsui/issues/1307 for TraitsUI minor release 7.1.0.

Release candidates
--------------------
- [x] Verify that all relevant PRs have been merged to master
- [x] Bump version number on master for continued development (#746)
- [x] From the commit just before bumping the version, create a new branch `maint/7.1`
- [x] Update `ci-src-requirements.txt` to target TraitsUI `maint/7.1` branch (#754)
- [x] Make sure `maint/7.1` is included (somehow) in the following block in appveyor.yml (#748, but see #751 as well. This step may not be needed in future releases)
```
branches:
  only:
    - ...
```
- [x] Update changelog and open PR targeting a new `maint/7.1` branch (#720)
- [x] Check MANIFEST and requirements are still up to date.
- [x] Update version in setup.py for the prerelease (#771)
       - (FTR: This step was done for 7.1.0rc2 but not 7.1.0rc1 due to #744.)
       - Create a new branch from maint/7.1: `git checkout maint/7.1; git pull; git checkout -b call-it-anything-you-like`
       - Set PRERELEASE to "rc2" and IS_RELEASED to true, commit, e.g.`git commit -m "Set IS_RELEASED to true for prerelease 7.1.0rc2"`
       - Flip IS_RELEASED back to false, commit.
       - Open a PR against `maint/7.1`, merge but DO NOT SQUASH
                No squashing so that the tagged commit in the next step will be on the branch (Kit: It is not a big deal to me even if the tagged commit was not on the branch. But some prefer it that way.) Alternatively, flip the IS_RELEASED flag in a separate PR.
- [x] Tag (annotated!) the release candidate on the commit where IS_RELEASED is set to true, e.g. `git tag -a -m "Release candidate 7.1.0rc2" 7.1.0rc2 <commit-hash>`
        - (FTR: the tag for 7.1.0rc1 was different due to #744)
- [x] Push the tag to GitHub, e.g. `git push origin 7.1.0rc2`
- [x] Upload to PyPI
      - (FTR: This step was done for 7.1.0rc2 but not 7.1.0rc1 due to #744.)
      - `git checkout <tag>`, `git clean -ffxd`, `python setup.py sdist`, `twine check dist/<...>.tar.gz`, `twine upload dist/<...>.tar.gz`
- [x] Announcement for the release candidate 

Release blockers
-----------------
- [x] #740 (#750 )
- [x] #739
- [x] #741 (#758)
- [x] #742 (#749)

Pre-release
---
- [x] Backport PRs that have been merged to master to the maintenance branch. Use the "need backport ..." tag for filtering (but don't rely 100% on it)(#777)
- [x] Update changelog for the backported PRs (#777)
- [x] Test against other ETS packages and other pyface-using projects
- [x] Check MANIFEST, requirements, changelog are still up to date.
- [x] Test building the documentation

Release
--------
- [x] Create branch `release/7.1.0` from `maint/7.1` branch. (#778)
        - Set release to `7.1.0`, and set `IS_RELEASED` is true; commit
        - Open a PR against `maint/7.1`  with this being the last commit so that CI is built on the release commit
        - Bump the version to `7.1.1` and set `IS_RELEASED` to false; commit, don't push yet.
        - Once CI is done building the last commit, push the commit
        - Merge but do not squash so that the release commit is on the branch
- [x] From the commit at which `IS_RELEASED` is true and version is `7.1.0`, tag (annotated!) `git tag -a -m "Release 7.1.0" 7.1.0`
- [x] Push the tag `git push origin 7.1.0`
- [x] After merge, restore the `release/7.1.0` branch (Kit: Not sure why we are keeping them, but we are it seems)
- [x] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting docs/index.html looks good. (#779)
- [x] Install from source distribution and run tests again
- [x] Upload to PyPI
- [x] Test the PyPI package

Post-release
-------------
- [x] Update GitHub Release pages https://github.com/enthought/pyface/releases
- [x] Package update for `enthought/free` repository (for EDM)
- [x] Backport release note and change log to master, and possibly `maint/7.1` branch (#780)
- [x] Announcement (e.g. Google Group)

**Comment by corranwebster** [comment]
The above needs some modification: you wrote

> This is a bug fix release and therefore the procedure will be slightly different compared to a major/minor release.
> See [enthought/traitsui#1307](https://github.com/enthought/traitsui/issues/1307) for TraitsUI minor release 7.1.0.

This is not a bugfix release.

**Comment by rahulporuri** [comment]
stupid copy-pasta error. thanks. corrected.

**Comment by rahulporuri** [comment]
I'm working on updating the changelog in preparation for the release.

**Comment by kitchoi** [comment]
For updating `gh-pages`, these are the commands I used:
```
git checkout 7.1.0
git clean —ffxd
python etstool.py install —toolkit=pyqt5
python etstool.py  docs —toolkit=pyqt5
git checkout gh-pages
git pull
mkdir release-7.0.1
mv *.html release-7.0.1/
mv api release-7.0.1/
mv objects.inv release-7.0.1/
mv _sources release-7.0.1/
mv _static release-7.0.1/
mv searchindex.js release-7.0.1/
rsync -r --exclude="\.buildinfo" docs/build/html/* .
git add *.html
git add _*
git add release-7.0.1
```

To verify:
```
python3.8 -m http.server 
```

**Comment by aaronayres35** [comment]
Closing as I just made the announcement which wraps up the issue checklist.
